### PR TITLE
Fixes #14079 - display_link_if_authorized not hidden when disabled

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -566,3 +566,17 @@ table {
   max-height: 200px;
   padding: 4px;
 }
+
+.dropdown-menu a[disabled],
```
Comment (tbrisker) [inline app/assets/stylesheets/application.scss]: Why all this CSS? `disabled` is not a valid attribute for a link at all - it does not prevent it from being clicked.

**Comment by tbrisker** [comment]
When do we actually use `disabled` with `display_link_if_authorized`? I only found one instance: https://github.com/theforeman/foreman/blob/develop/app/views/hosts/edit.html.erb#L7
but I may have missed other places.
Since `disabled` is not a valid attribute for a link, and I'm guessing this check is a leftover from a very old time when we didn't have the permissions yet. I think we should just remove the check and fix any place that may depend on it so that the behaviour is correct.

**Comment by xprazak2** [comment]
I removed the check, now the behaviour is as follows:

when not authorized, the link/button is not displayed
when authorized and disabled, the link/button is disabled
when authorized, the link/button is visible and enabled

Using `disabled` for [links](https://github.com/theforeman/foreman/blob/develop/app/helpers/hosts_helper.rb#L298) seems to be common.

**Comment by xprazak2** [comment]
[test]

**Comment by tbrisker** [comment]
@xprazak2 Ahha, I found the reason disabled links are disabled: https://github.com/theforeman/foreman/blob/develop/app/assets/javascripts/application.js#L101 
I sent you a PR with a suggestion to simplify the check and maintain the current behaviour: https://github.com/xprazak2/foreman/pull/1

**Comment by xprazak2** [comment]
I merged the simplified check.

**Comment by tbrisker** [comment]
@xprazak2 tests are failing, looks like there is a missing `)` or extra `,` or something somewhere

**Comment by tbrisker** [comment]
@orrabin Care to take over the review for this?

**Comment by orrabin** [comment]
merged as a60057600ed79d7d8b1b3123f6c9031b57272e78, thanks @xprazak2 and @tbrisker for reviewing

**Pull request closed without merging**